modules/ssh/ssh.go

```diff
@@ -217,6 +326,65 @@ func FetchContentsOfFileE(t *testing.T, host Host, useSudo bool, filePath string
 	return CheckSshCommandE(t, host, command)
 }
 
+func listFileInRemoteDir(t *testing.T, sshSession *SshSession, options ScpDownloadOptions) ([]string, error) {
+	logger.Logf(t, "Running command %s on %s@%s", sshSession.Options.Command, sshSession.Options.Username, sshSession.Options.Address)
+
+	var result []string
+
+	if err := setUpSSHClient(sshSession); err != nil {
+		return result, err
+	}
+
+	if err := setUpSSHSession(sshSession); err != nil {
+		return result, err
+	}
+
+	findCommand := fmt.Sprintf("find %s -type f", options.RemoteDir)
+
+	if options.FileNameFilter != "" {
+		findCommand = fmt.Sprintf("%s -name %s", findCommand, options.FileNameFilter)
+	}
+
+	if options.MaxFileSizeMB != 0 {
+		findCommand = fmt.Sprintf("%s -size -%dM", findCommand, options.MaxFileSizeMB)
+	}
+
+	r, err := sshSession.Session.Output(findCommand)
+
+	if err != nil {
+		return result, err
+	}
+	defer sshSession.Session.Close()
+
+	resultString := string(r)
+	resultString = resultString[:len(resultString)-1]
+
+	result = append(result, strings.Split(resultString, "\n")...)
+	return result, nil
```

Comment:
Should the return type be a map from filename to string?

Reply:
What will the `string` be? Right now, I'm just returning the full path of the matching files on the remote machine.

Reply:
Oh, my bad, I misread this as though these were the contents of the files rather than the paths.